### ISSUE TITLE
[Accessibility] Accessibility / 508 compliance ?

### ISSUE THREAD

**Issue by wesleyahall** [facebook/react-native#342]
Movie ListView Example :  iOS Accessibility (Screen Reader) is not reading anything, even the plain text elements on the screen. The images are skipped entirely, as if they were background images or something. 

Is there a way to get accessibility / screen reader mode activated? 
If not, is it planned (soon)? 
Or... do I have to go back to cordova? 

This is a 100% breaking issue for my company. I guess I was dumb to assume "native" would include accessibility support.

**Comment by frantic** [comment]
We definitely do support that. See https://github.com/facebook/react-native/blob/master/Libraries/Components/View/View.js#L69
I know @vjeux implemented the support in ObjC land that automagically makes UI accessible, but maybe his change was lost during refactoring. cc @nicklockwood

**Comment by wesleyahall** [comment]
I figured out the image problem (read the docs 😑) by adding accessible={true} but this prop doesn't appear to be available on Text components. How do we make the non-touchable Text components still readable?

**Comment by vjeux** [comment]
All the text elements should be accessible by default. Need to ensure that the accessibility features that we have in the internal codebase are ported to open source.

**Comment by hapasa** [comment]
There appears to be code looking for accessiblityLabel from the UIView.
Is there a way to set that from javascript?

**Comment by hapasa** [comment]
Ok, the following works: <Text accessibilityLabel='Hey, I'm accessible!'>Hello World</Text>
But accessibilityLabel should default to 'Hello World' in the above.

**Comment by ide** [comment]
0.5.0-rc includes accessibility improvements: https://github.com/facebook/react-native/releases/tag/v0.5.0-rc

Can someone familiar with accessibility report whether the improvements are good enough and open up more specific issues if there are areas that still need to be addressed?

**Comment by brentvatne** [comment]
ping @wesleyahall

**Comment by christopherdro** [comment]
@wesleyahall  Any updates on this issue?

**Comment by brentvatne** [comment]
I'm using accessibility features extensively on an app to integrate with KIF for acceptance testing, everything seems to be working well. If you stumble upon this and accessibility tools within React Native aren't currently meeting your needs, *please* open an issue and let us know what your use cases are! Thanks!
